hdf5-sys/build.rs

```diff
@@ -634,6 +641,24 @@ impl Config {
             println!("cargo:have_threadsafe=1");
         }
     }
+
+    fn check_against_features_required(&self) {
+        if feature_enabled("DEPRECATED") {
+            assert!(!self.header.have_no_deprecated, "Required deprecated symbols are not present")
+        }
+        if feature_enabled("THREADSAFE") {
+            assert!(
+                self.header.have_threadsafe,
+                "Required threadsafe but library was not build using the threadsafe option"
+            );
+        }
+        if feature_enabled("ZLIB") {
+            assert!(
+                self.header.have_zlib,
+                "Required zlib filter but library does not have builtin support for this options"
+            );
+        }
```

Comment:
Maybe something like this for consistency of error messages?
```rust
let h = &self.header;
for (flag, feature, native) in [
    (h.have_no_deprecated, "no-deprecated", "H5_NO_DEPRECATED_SYMBOLS"),
    (h.have_threadsafe, "threadsafe", "H5_HAVE_THREADSAFE"),
    (h.have_zlib, "zlib", "H5_HAVE_FILTER_DEFLATE"),
] {
    if feature_enabled(feature.to_ascii_uppercase().replace('-', '_')) {
        assert!(flag, "Enabled feature {feature:?} but the HDF5 library wasn't built with {native}");
    }
}

Reply:
I used your suggestions but instead use the compile time flags which are used for building HDF5 with the feature enabled

Reply:
> I used your suggestions but instead use the compile time flags which are used for building HDF5 with the feature enabled

That's even better, thanks! (I think that's what I meant really)